### ISSUE TITLE
Autoformatting breaks Dockerfile with heredoc

### ISSUE THREAD

**Issue by wedi** [microsoft/vscode-docker#3327]
Using the new heredoc syntax in Dockerfiles used to trigger error messages in VS Code but autoformatting did not interfere with it. Unfortunately this has changed (unfortunately I cannot tell when): since some time autoformatting indents the end marker and following lines which makes it impossible to use autoformatting on save.

This dockerfile

```dockerfile
# syntax=docker/dockerfile:1.3-labs
FROM alpine
RUN <<EOF
    echo "Heredocs! Finally!"
EOF
CMD ["sh", "-c", "sleep 5"]
```

becomes

```dockerfile
# syntax=docker/dockerfile:1.3-labs
FROM alpine
RUN <<EOF
    echo "Heredocs! Finally!"
    EOF
    CMD ["sh", "-c", "sleep 5"]
```

 after formatting.

**Comment by rcjsuen** [comment]
@wedi Thanks for the bug report.

For a workaround, you can set the `"docker.languageserver.formatter.ignoreMultilineInstructions"` preference to `true`. See #2976.

**Comment by wedi** [comment]
Thanks! I definitely prefer fixing multiline commands myself over fixing the dockerfile in an external editor. :)

**Comment by bwateratmsft** [comment]
@rcjsuen do you want to create an issue in https://github.com/rcjsuen/dockerfile-language-server-nodejs for this? Or keep it here?

**Comment by rcjsuen** [comment]
> @rcjsuen do you want to create an issue in https://github.com/rcjsuen/dockerfile-language-server-nodejs for this?

I've opened https://github.com/rcjsuen/dockerfile-utils/issues/105 to track this.

**Comment by bwateratmsft** [comment]
Sounds good. Since there's a workaround and we pick up the latest language server regularly I'll close this one.

**Comment by bwateratmsft** [comment]
This is now released in Docker extension version 1.19.0.
